pytorch_lightning/loops/optimization/optimizer_loop.py

```diff
@@ -369,7 +369,7 @@ def _optimizer_step(
             optimizer,
             opt_idx,
             train_step_and_backward_closure,
-            on_tpu=(self.trainer._device_type == _AcceleratorType.TPU and _TPU_AVAILABLE),
+            on_tpu=isinstance(self.trainer.accelerator, TPUAccelerator),
```

Comment:
We still need the _TPU_AVAILABLE check here. We haven't enable accelerator.is_available check yet. It's one of the follow up items. A lot of tests fails when we enable it, so didn't include in the big PR

Reply:
When would `TPUAccelerator` be selected without TPU available?

Reply:
yeah, self.trainer.accelerator = TPUAccelerator implies that _TPU_AVAILABLE, otherwise the acc connector wouldn't select it in the first place.